nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement the correct_name method'
```

Comment:
- [ ] **[OPTIONAL]** You can make this error message more precise to point to the class where the method is not implemented.

  ```
  raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
  ```